File: humai/sendtoken.py
import os
from api.cline_trans import Cline_trans
from api.keys_trans import INRKey
from ineryjs import Api, JsonRpc, RpcError, JsSignatureProvider

os.environ.update({k.lower(): v for k, v in os.environ.items()})

TOKEN_CONTRACT = os.getenv("TOKEN_CONTRACT")
SEND_ACCOUNT = os.getenv("INERY_ACCOUNT")
TOKEN_SYMBOL = os.getenv("TOKEN_SYMBOL")
SEND_QUANTITY = os.getenv("SEND_QUANTITY")
SEND_MEMO = os.getenv("SEND_MEMO")

json_rpc = JsonRpc(os.getenv("INERY_NODE_URL"))
signature_provider = JsSignatureProvider([os.getenv("PRIVATE_KEY")])

api = Api({
    "rpc": json_rpc,
    "signature_provider": signature_provider
})

async def SEND_token(token_contract, SEND, quantity, memo):
    try:
        tx = await api.transact({
            "actions": [
                {
                    "account": TOKEN_CONTRACT,
                    "name": "SEND",
                    "authorization": [
                        {
                            "actor": SEND_ACCOUNT,
                            "permission": "active"
                        }
                    ],
                    "data": {
                        "quantity": quantity,
                        "memo": memo
                    }
                }
            ]
        }, {"broadcast": True, "sign": True})

        print(tx["processed"])
        print(tx["processed"]["action_traces"][0]["act"])
    except RpcError as error:
        print(f"ERROR: Can't SEND token symbol {TOKEN_SYMBOL}")
        print(f"DETAILS: {error.details[0]['message']}")

SEND_token( TOKEN_CONTRACT , SEND_ACCOUNT , SEND_QUANTITY , SEND_MEMO )

File: kriptzy/sendToken.ts
import { Api, JsonRpc, JsSignatureProvider } from "ineryjs";
import dotenv from "dotenv";
dotenv.config();

const json_rpc = new JsonRpc(process.env.URL_NODE as string);
const account = "kriptzy";
const actor = "kriptzy";
const privateKey = process.env.PRIVATE_KEY;
const signature = new JsSignatureProvider([privateKey!]);
const api = new Api({
  rpc: json_rpc,
  signatureProvider: signature,
});

interface IProps {
  from: string;
  to: string;
  quantity: string;
  memo: string;
}

const sendToken = async ({ from, to, quantity, memo }: IProps) => {
  try {
    const tx = await api.transact(
      {
        actions: [
          {
            account,
            name: "transfer",
            authorization: [
              {
                actor,
                permission: "active",
              },
            ],
            data: {
              from,
              to,
              quantity,
              memo,
            },
          },
        ],
      },
      { broadcast: true, sign: true }
    );
    console.log(tx);
  } catch (error) {
    console.log(error);
  }
};

sendToken({
  from: "kriptzy",
  to: "inery",
  quantity: "10 INR",
  memo: "inery task 4",
});
